src/kernels/kernelStatusProvider.ts

```diff
@@ -0,0 +1,68 @@
+// Copyright (c) Microsoft Corporation.
+// Licensed under the MIT License.
+
+import { inject } from 'inversify';
+import { IExtensionSyncActivationService } from '../platform/activation/types';
+import { disposeAllDisposables } from '../platform/common/helpers';
+import { IDisposable, IDisposableRegistry } from '../platform/common/types';
+import { DataScience } from '../platform/common/utils/localize';
+import { KernelProgressReporter } from '../platform/progress/kernelProgressReporter';
+import { IStatusProvider } from '../platform/progress/types';
+import { getDisplayNameOrNameOfKernelConnection } from './helpers';
+import { IKernel, IKernelProvider } from './types';
+
+export class KernelStatusProvider implements IExtensionSyncActivationService {
```

Comment:
This simple change reduces the coupling and the number of arguments passed all around